x/gov/client/rest/rest.go

```diff
@@ -0,0 +1,248 @@
+package rest
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/wire"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/tendermint/go-crypto/keys"
+)
+
+// RegisterRoutes - Central function to define routes that get registered by the main application
+func RegisterRoutes(ctx context.CoreContext, r *mux.Router, cdc *wire.Codec, kb keys.Keybase) {
+	r.HandleFunc("/gov/submitproposal", postProposalHandlerFn(cdc, kb, ctx)).Methods("POST")
```

Comment:
Proposal for rest-ful structure:

submit: POST /gov/proposals
deposit: POST /gov/proposals/{proposalID}/deposits
vote: POST /gov/proposals/{proposalID}/votes
get votes for address: GET /gov/proposals/{proposalID}/votes?address={voterAddress}
read proposal: GET /gov/proposals/{proposalID}

let me know what you think

Reply:
Extracted into https://github.com/cosmos/cosmos-sdk/issues/1284